### PR TITLE
enterprise-portal: initialize service with dotcom DB connection

### PR DESCRIPTION
**Pull request by bobheadxi**
Closes https://linear.app/sourcegraph/issue/CORE-96 - this scaffolds an initial Enterprise Portal service, as described in [RFC 885](https://docs.google.com/document/d/1tiaW1IVKm_YSSYhH-z7Q8sv4HSO_YJ_Uu6eYDjX7uU4/edit#heading=h.tdaxc5h34u7q), with a test setup to verify that we are able to connect to the Sourcegraph.com database and query from it. This will be used to implement the read-only RPCs proposed in #62263 

## Test plan

- [x] Publish image by hand:
  ```sh
  bazel run //cmd/enterprise-portal:candidate_push --stamp -- --tag insiders --repository us.gcr.io/sourcegraph-dev/enterprise-portal
  ```
- [x] Deploy in an MSP environment with appropriate permissions configured on the dotcom database, verify connection works: https://github.com/sourcegraph/managed-services/pull/1368 https://github.com/sourcegraph/managed-services/pull/1377

### CODE REVIEW

**Comment by bobheadxi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/62525?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#62525** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/62525?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#62524** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/62524?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bobheadxi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cmd/enterprise-portal/service/dotcomdb.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	"github.com/jackc/pgx/v5"
+
+	"github.com/sourcegraph/sourcegraph/cmd/enterprise-portal/internal/dotcomdb"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/lib/managedservicesplatform/cloudsql"
+)
+
+func newDotComDBConn(ctx context.Context, config Config) (*dotcomdb.Database, error) {
+	// TODO allow override
```
Comment (unknwon) [inline cmd/enterprise-portal/service/dotcomdb.go:14]: TODO implemented already?

**Review by unknwon — APPROVED** [review]

File: cmd/enterprise-portal/service/dotcomdb.go (hunk 1 above)
Comment (bobheadxi) [reply cmd/enterprise-portal/service/dotcomdb.go:14]: ah yes, it's right there 😁

Comment (bobheadxi) [reply cmd/enterprise-portal/service/dotcomdb.go:14]:
```suggestion
```

**Pull request merged**